### DISCUSSION TITLE
Getting self.NO_VALID_SERVER_VERSIONS(f"Tried versions {self._api_versions_to_try()} but all were invalid") error

### DISCUSSION THREAD

**Discussion by prashanttest-elastic** [ecederstrand/exchangelib#1303] [Q&A]
while updating my application with latest version of exchangelib 5.3.0 i am getting error while fetching mails from exchange server - 
```
self.NO_VALID_SERVER_VERSIONS(f"Tried versions {self._api_versions_to_try()} but all were invalid")
exchangelib.errors.ErrorInvalidSchemaVersionForMailboxVersion: Tried versions ('Exchange2016', 'Exchange2015_SP1', 'Exchange2015', 'Exchange2013_SP1', 'Exchange2013', 'Exchange2010_SP2', 'Exchange2010_SP1', 'Exchange2010', 'Exchange2007_SP1', 'Exchange2007') but all were invalid
```
with older version (5.0.3) it is working fine for exchange server but due to `ErrorAccessDenied: Not allowed to access Non IPM folder` error in office365 i am upgrading to 5.3.0 and now i am getting No valid server version error. 

Can someone help me with this issue.

**Comment by ecederstrand** [comment]
I believe this was also reported in https://github.com/ecederstrand/exchangelib/issues/1301
